src/rules/prettier.ts

```diff
@@ -36,12 +36,36 @@ export default {
     return {
       Program() {
         const sourceCode = context.sourceCode.text
-        const formatted = format(sourceCode, {
-          filepath: context.filename,
-          ...(context.options[0] || {}),
-        })
+        try {
+          const formatted = format(sourceCode, {
+            filepath: context.filename,
+            ...(context.options[0] || {}),
+          })
 
-        reportDifferences(context, sourceCode, formatted)
+          reportDifferences(context, sourceCode, formatted)
+        }
+        catch (error) {
+          let loc = {
+            start: { line: 1, column: 0 },
+            end: { line: 1, column: 0 },
+          }
+
+          if (error instanceof Error) {
+            const locationMatch = error.message.match(/\((?<line>\d):(?<column>\d)\)/)
+            if (locationMatch?.groups) {
+              const { line, column } = locationMatch.groups
+              loc = {
+                start: { line: +line, column: +column },
+                end: { line: +line, column: +column },
+              }
+            }
+          }
+
+          context.report({
+            loc,
+            message: 'Failed to format the code',
+          })
```

Comment:
I'd recommend you to look into this: [eslint-plugin-prettier handling errors thrown by Prettier](https://github.com/prettier/eslint-plugin-prettier/blob/de9751c85d059678904035322501bfce120f61b7/eslint-plugin-prettier.js#L209-L233)
(Actually I was going to submit a PR with these changes and just found this PR :joy:)

Reply:
Updated to use the same logic with `eslint-plugin-prettier`. Thank you!